src/Compilers/Core/Portable/SourceGeneration/Nodes/NodeStateTable.cs

```diff
@@ -124,61 +130,63 @@ public ImmutableArray<T> Batch()
             return sourceBuilder.ToImmutableAndFree();
         }
 
-        public Builder ToBuilder(int? extraCapacity = null)
+        public Builder ToBuilder()
         {
             Debug.Assert(!this.IsFaulted);
-            int? capacity = extraCapacity.HasValue ? this._states.Length + extraCapacity.Value : null;
-            return new Builder(this, capacity);
+            return new Builder(this, copyPrevious: false);
         }
 
         public static NodeStateTable<T> FromFaultedTable<U>(NodeStateTable<U> table)
         {
             Debug.Assert(table.IsFaulted);
-            return new NodeStateTable<T>(Empty._states, isCompacted: true, table._exception);
-        }
-
-        public static NodeStateTable<T> WithSingleItem(T item, EntryState state)
-        {
-            return new NodeStateTable<T>(ImmutableArray.Create(ImmutableArray.Create((item, state))), isCompacted: false, exception: null);
+            return new NodeStateTable<T>(Empty._states, isCompacted: false, table._exception);
```

Comment:
>false [](http://example.com/codeflow?start=69&length=5)

I assume this could be `false` or `true` if there are no states. Is there a reason to use `false`?

Reply:
An invariant that a faulted table can't be compacted. I'll add an assert in the ctor to enforce it.

---
In reply to: [632134134](https://github.com/dotnet/roslyn/pull/53383#discussion_r632134134) [](ancestors = 632134134)